xsd/parse.go

```diff
@@ -140,20 +176,33 @@ func Parse(docs ...[]byte) ([]Schema, error) {
 	for _, root := range schema {
 		tns := root.Attr("", "targetNamespace")
 		s := Schema{TargetNS: tns, Types: make(map[xml.Name]Type)}
+		sHash := hash(root)
+
 		if err := s.parse(root); err != nil {
 			return nil, err
 		}
-		parsed[tns] = s
```

Comment:
Storing generated schemas by tns causes collisions, retaining only the types from the last schema to be generated targeting the specific namespace. Using a hash of the schema ensures all unique schemas are retained.

Reply:
This isn't the approach I would have used. Being able to assume that your schema document is exhaustive for its target namespace makes later steps simpler. For example, if there was a 1:1 mapping from target namespace to xsd then you could remove one layer of maps for your `prepCopyEltNamesToAnonTypes`, and you would be able to push it into the `CopyEltNamesToAnonTypes` function.

To workaround the pesky XSD specification allowing schema to be split across multiple files you could add a pass that merges all *xmltree.Element trees for schema with the same target namespace. Another benefit of doing the transform on the tree structure is that it's easier to verify, by printing out the XML in intermediate steps. What do you think?

Reply:
will think on this